File: scripts/gen_buildings.pl
#!/usr/bin/perl
# Made by kk

use strict;
use warnings;

use lib '.';
use build;
use gen;

gen->read_level();

gen->overlay_anywhere(build::get_building(rand(13)+6, rand(13)+6), 2, ',');
gen->overlay_anywhere(build::get_building(rand(13)+6, rand(13)+6), 2, ',');
gen->overlay_anywhere(build::get_building(rand(13)+6, rand(13)+6), 2, ',');
gen->overlay_anywhere(build::get_building(rand(13)+6, rand(13)+6), 2, ',');
gen->overlay_anywhere(build::get_building(rand(13)+6, rand(13)+6), 2, ',');

gen->print_level();

File: scripts/gen_buildings.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#include "array.h"
#include "gen.h"
#include "build.h"


int main(int argc, char *argv[]) {
	
	array_t* T = read_level(-1);
	srand(time(NULL));
	
	overlay_anywhere(get_building(rand() % 13 + 6, rand() % 13 + 6), 2, ',');
	overlay_anywhere(get_building(rand() % 13 + 6, rand() % 13 + 6), 2, ',');
	overlay_anywhere(get_building(rand() % 13 + 6, rand() % 13 + 6), 2, ',');
	overlay_anywhere(get_building(rand() % 13 + 6, rand() % 13 + 6), 2, ',');
	overlay_anywhere(get_building(rand() % 13 + 6, rand() % 13 + 6), 2, ',');
	
	print_level(-1);
	
	free_array(T);
	free(_get_world_ref());
	
	return 0;
}
